RGB LED Control V1.0/code/APP/app.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * 	File:						app.c
+ *	Module:					APP
+ *	Description: 		source file 
+ * 	Author: 				HAZEM ASHRAF
+ */ 
+
+/*=========FILE INCLUSION=========*/
+#include "app.h"
+
+/*=========MACRO DEFINITIONS=========*/
+#define NUMBER_OF_ITERATIONS_PER_ONE_MILI_SECOND 762
+#define MAX_PRESS 5
+#define RESET 0
+/*=========EXTERN VARIABLES=========*/
+
+/*=========GLOBAL VARIABLES=========*/
+
+/*=========TYPE DEFINITIONS=========*/
+
+/*=========APIs DECLERATIONS=========*/
+
+void Delay_MS(unsigned long long n)
+{
+    volatile unsigned long long count = 0;
+    while(count++ < (NUMBER_OF_ITERATIONS_PER_ONE_MILI_SECOND * n) );
+}
+
+void app_init(void)
+{
+	LED_init(ENU_PORT_F, ENU_PIN_1);
+	LED_init(ENU_PORT_F, ENU_PIN_2);
+	LED_init(ENU_PORT_F, ENU_PIN_3);
+	BUTTON_init(ENU_PORT_F, ENU_PIN_0);
+}
+
+void app_start(void)
+{
+	uint8 button_value	= 0;
+	boolean flag				= TRUE;
+	uint8 press_count		= RESET;
+	/* Superloop */
+  while(1)
+	{
+		BUTTON_digitalRead(ENU_PORT_F,ENU_PIN_0,&button_value);
+		if(button_value == LOGIC_HIGH)
+		{
+			Delay_MS(50);
+			BUTTON_digitalRead(ENU_PORT_F,ENU_PIN_0,&button_value);
+			if(button_value == LOGIC_HIGH)
+			{
+				if(flag == FALSE)
+				{
+					press_count++;
+					switch (press_count)
+					{
```

Comment:
switch case should only have one break.

Reply:
how to to use switch case with only one brake ?

Reply:
I think there are two solutions.
1. The easy way to replace the switch case with
2. The hard way is to use Fall-through switch cases.

Reply:
Done, you can check it now

Reply:
is there any refernce for the rule regard switch case shuld have only one break statement?

Reply:
The only available reference until now is our website